### ISSUE TITLE
Already trained weights - adding more samples and train further with that weight file?

### ISSUE THREAD

**Issue by flowzen1337** [AlexeyAB/darknet#5028]
Hi @AlexeyAB ,

so I've trained a weight file for 500k iterations with 36 classes. Most of the classes are working brilliant ( thumbs up for your work on yolo! :-) ) but a handfull of these 36 classes doesnt detect all the objects (i guess thats because of not enough samples for that classes).

So my question is, can i use the already trained weight file, add more samples to my training sample-set for some classes and continue training it with the now more samples? That would help a lot as i dont have to start the training for all classes from zero.

Is it as easy as the normal way with training:
- Creating the training/testing sample-set with the corresponding *.txt files
- Just adding the new image samples to the training/test files
- And adding the already pre-trained *.weight file to the training command as the last checkpoint?

So for understanding that's the way i guess i want to go:

` ./darknet detector train "data/model_name/mymodel.data" "cfg/mymodel.cfg"  darknet53.conv.74 -dont_show -map -gpus 0,1,2,3`
...
...
500k iterations later checked the weight file see that some objects are not so good detected and i want to add more samples, can i just add the samples and run the train task again with giving the last weight file like this?
` ./darknet detector train "data/model_name/mymodel.data" "cfg/mymodel.cfg" "weights/mymodel_last.weights" darknet53.conv.74 -dont_show -map -gpus 0,1,2,3`

Or do i have sadly start from zero again? 
Thanks for your answer in advance!

**Comment by AlexeyAB** [comment]
> So my question is, can i use the already trained weight file, add more samples to my training sample-set for some classes and continue training it with the now more samples?

Yes, you can.

> And adding the already pre-trained *.weight file to the training command as the last checkpoint?

> 500k iterations later checked the weight file see that some objects are not so good detected and i want to add more samples, can i just add the samples and run the train task again with giving the last weight file like this?
> ./darknet detector train "data/mymodel.data" "cfg/mymodel.cfg" "weights/mymodel_last.weights" darknet53.conv.74 -dont_show -map -gpus 0,1,2,3

Yes.
Or increase max_batches= in your cfg-file.
Or run training with flag `-clear`

**Comment by flowzen1337** [comment]
Thanks @AlexeyAB !

What does the "-clear" flag do? Or is there any doc/readme where i can see all the possible parameters for "./darknet train... ? 
On the README section i couldnt find anything for the -clear flag :(

Thanks again mate!

**Comment by AlexeyAB** [comment]
It just set trained iterations counter = 0

There is wiki, but there are dsecribed only parameters of cfg files https://github.com/AlexeyAB/darknet/wiki

**Comment by flowzen1337** [comment]
Thanks again Mister :-)

**Comment by jfkkf123** [comment]
> ` ./darknet detector train "data/mymodel.data" "cfg/mymodel.cfg" "weights/mymodel_last.weights" darknet53.conv.74 -dont_show -map -gpus 0,1,2,3`

hello, i confuse why you use  ''darknet53.conv.74'' here? and when you continue training your network, the training-set is only new image samples or  adding new images to original training-set?
thanks

**Comment by flowzen1337** [comment]
> > ` ./darknet detector train "data/mymodel.data" "cfg/mymodel.cfg" "weights/mymodel_last.weights" darknet53.conv.74 -dont_show -map -gpus 0,1,2,3`
> 
> hello, i confuse why you use ''darknet53.conv.74'' here? and when you continue training your network, the training-set is only new image samples or adding new images to original training-set?
> thanks

Because it is described in the README section:

https://github.com/AlexeyAB/darknet#how-to-train-to-detect-your-custom-objects ( Bulletpoint 7. and 8. - I've used yolo3.cfg - therefore darknet53.conv.74) or am i wrong and i shouldnt use it?

I've added the new images/annotations on top of the original dataset, but currently it seems (after 100.000 iterations that it dont have any impact and i have the same results as before without the new annotations/images :-( )

**Comment by kepler62f** [comment]
@flowzen1337 
I was researching on this topic and came across your issue. Just curious, when you retake training with your previous weight file and new data, did you also have mAP figures?  When you start the new training, do the mAP figures etc. show levels similar to those when you stopped in previous training?

**Comment by tonginleong** [comment]
I am in a similar situation where I have a trained weighted with my old dataset (1 class, 200k iterations with 64 batch_size, 200k images)

I want to add 200k images (completely new) to this model to ramp up its accuracy/make it more generic/susceptible to real life changes, do I put old 200k + new 200k=400k images to retrain, or just the new 200k images? (and will be repeated in the future to improve the model).

and of course I know I will need to change the cfg accordingly and I am aware of the tags and other issue but they dont give me a clear explanation of this.
